src/main/java/org/skriptlang/skript/bukkit/damagesource/elements/ExprCausingEntity.java

```diff
@@ -0,0 +1,90 @@
+package org.skriptlang.skript.bukkit.damagesource.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.damage.DamageSource;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.damagesource.DamageSourceExperiment;
+import org.skriptlang.skript.bukkit.damagesource.DamageSourceWrapper;
+import org.skriptlang.skript.bukkit.damagesource.elements.ExprSecDamageSource.DamageSourceSectionEvent;
+
+@Name("Damage Source - Causing Entity")
+@Description({
+	"The causing entity of a damage source.",
+	"The causing entity is the entity that ultimately caused the damage. (e.g. the entity that shot an arrow)",
+	"When setting a 'causing entity' you must also set a 'direct entity'.",
+	"Cannot change any attributes of a damage source outside the 'custom damage source' section."
+})
+@Example("""
+	set {_source} to a new custom damage source:
+		set the damage type to magic
+		set the causing entity to {_player}
+		set the direct entity to {_arrow}
+		set the damage location to location(0, 0, 10)
+	""")
+@Example("""
+	on damage:
+		set {_causing} to the causing entity of event-damage source
+	""")
+@Since("INSERT VERSION")
+@RequiredPlugins("Minecraft 1.20.4+")
+public class ExprCausingEntity extends SimplePropertyExpression<DamageSource, Entity> implements DamageSourceExperiment {
+
+	static {
+		registerDefault(ExprCausingEntity.class, Entity.class, "causing entity", "damagesources");
```

Comment:
it's very unambiguous, which is good, but `causing entity` just doesn't really make much grammatical sense to me. An alternative like `responsible entity` might be good to add